src/parsers/MetadataParser.ts

```diff
@@ -95,29 +115,44 @@ export default class MetadataParser extends StatefulBufferParser {
     const mapMetadata = this.parseEncodedMapMetaString(
       this.decodeGameMetaString(encodedString),
     );
-    this.skip(12);
+    const playerCount = this.readUInt32LE();
+    const gameType = this.readStringOfLength(4, "hex");
+    const localeHash = this.readStringOfLength(4, "hex");
     const playerListFinal = playerRecords.concat(
       playerRecords,
       this.parsePlayerList(),
     );
     let reforgedPlayerMetadata: ReforgedPlayerMetadata[] = [];
-    if (this.readUInt8() !== 25) {
-      this.skip(-1);
+    if (this.peekUInt8() !== 25) {
       reforgedPlayerMetadata = this.parseReforgedPlayerMetadata();
     }
-    this.skip(2);
+    if (this.readUInt8() !== 25) {
+      console.error("Unknown chunk detected!");
```

Comment:
Instead of using two log statements, can we use one and just concat the buffer to the string? Just use + this.buffer.subarray(...).toString("hex")